rust-sdk/src/client.rs

```diff
@@ -375,10 +328,15 @@ impl<'a> Client<'a> {
         subject_attributes: &Attributes,
         expected_type: Option<VariationType>,
         convert: impl FnOnce(AssignmentValue) -> T,
-    ) -> Result<Option<T>> {
+    ) -> Result<Option<T>, FlagEvaluationError> {
         let config = self.configuration_store.get_configuration();
-        let assignment =
-            config.get_assignment(flag_key, subject_key, subject_attributes, expected_type)?;
+        let assignment = get_assignment(
```

Comment:
I don't think we want the "untyped" assignment methods in UFC, forcing folks to declare the type they want.

Reply:
`get_assignment` here is an internal function.

We do have a top-level `Client.get_assignment` and `Client.get_assignment_details` though, that I'm OK to remove